openstudiocore/src/openstudio_lib/SpacesSubtabGridView.cpp

```diff
@@ -866,8 +867,12 @@ namespace openstudio {
 
   void SpacesSubtabGridView::purgeObjects(const IddObjectType& iddObjectType)
   {
-    for (auto mo : this->m_model.getConcreteModelObjects<model::Space>()){
-      mo.remove();
+    // Purge Spaces. The logic is to remove spaces that don't have a thermalZone nor surfaces
+    for (auto mo : this->m_model.getConcreteModelObjects<model::Space>())
```

Comment:
@jmarrec this seems like reasonable logic, but I wonder if there are any other things that should be added?  shading groups?

Reply:
Also should thermal zone be considered?  What about building story?

Reply:
Seems like it would be a good idea to add an API method to all objects that was "used" or something, that way you could see which objects were going to be purged before you did it.  Oh well, sometime later.

Reply:
I think the real key here is geometry. The only thing a space must have is geometry.
I'm not sure whether you'd add a space with just shading?
If not, then the space should be used by a ThermalZone, because E+ doesn't have the concept of space, it ends up in a zone, no?